WordPress/Classes/Services/BlogService.m

```diff
@@ -671,6 +671,17 @@ - (void)mergeBlogs:(NSArray<RemoteBlog *> *)blogs withAccount:(WPAccount *)accou
         [self updateBlogWithRemoteBlog:remoteBlog account:account];
     }
 
+    /*
+     Sometimes bad things happen and blogs get duplicated. 👭
+     Hopefully we'll fix all the causes and this should never happen again 🤞🤞🤞
+     But even if it never happens again, it has already happened so we need to clean up. 🧹
+     Otherwise, users would have to reinstall the app to get rid of duplicates 🙅‍♀️
+
+     More context here:
+     https://github.com/wordpress-mobile/WordPress-iOS/issues/7886#issuecomment-524221031
+     */
+    [self deduplicateBlogsForAccount:account];
```

Comment:
Should we do it on every `syncBlogsForAccount:` call? Seems like once during an app startup _should_ be enough?

Reply:
We don't really know for sure how blogs get duplicated, and in any case I think it belongs conceptually in the sync method: when you call sync you expect the local list to reflect what's on the server.

The implementation will bail early if it detects no duplicates, so it shouldn't be too much overhead.

I wouldn't mind revisiting this once we have all the other fixes in place though.